### PR TITLE
SE-465: Fix mixed lookthrough instrument upload

### PR DESCRIPTION
**Pull request by thomasdotred**
SE-465: Initial fix

SE-465: Add tests for lookthrough instruments upsert
Allow non lookthrough and lookthrough instruments to be upserted in the same dataframe.

# Pull Request Checklist

- [x] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [X] Tests pass locally

# Description of the PR

Allows load from data frame to upsert a mixture of conventional and lookthrough instruments

### CODE REVIEW

File: lusidtools/cocoon/utilities.py [hunk 1]
```diff
@@ -254,6 +254,7 @@ def set_attributes_recursive(
     # Used to check if all attributes are none
     total_count = 0
     none_count = 0
+    missing_value = False
```
Comment (rizwansaeed) [inline lusidtools/cocoon/utilities.py:257]: do we need some tests specifically around these changes too?

**Review by rizwansaeed — CHANGES_REQUESTED** [review]

File: tests/integration/cocoon/test_cocoon_instruments.py [hunk 2]
```diff
@@ -1126,3 +1124,101 @@ def test_load_instrument_lookthrough_portfolio(self):
...
+                    "required": {"name": "instrument_name"},
+                    "optional": {
+                        "look_through_portfolio_id.scope": "$test-mixed-lookthrough",
+                        "look_through_portfolio_id.code": "lookthrough_code",
+                    },
+                },
+            ],
+        ]
+    )
+    def test_load_instrument_lookthrough(self, _, df, mapping):
```
[the remark this answers is not in the document]
Comment (thomasdotred) [reply tests/integration/cocoon/test_cocoon_instruments.py:1172]: same instrument id will be covered by instrument upsert tests that throw errors if multiple instruments of the same type are included in the same dataframe
image

File: tests/integration/cocoon/test_cocoon_instruments.py [hunk 3]
```diff
@@ -1126,3 +1124,101 @@ def test_load_instrument_lookthrough_portfolio(self):
...
+        code = create_scope_id()
```
[the remark this answers is not in the document]
Comment (thomasdotred) [reply tests/integration/cocoon/test_cocoon_instruments.py]: added teardown for this test. 
Maybe in the future we should move towards tagging all scopes with some prefix, then have a generic teardown that runs after all unit tests and deletes every related portfolio, property definition, quote, recipe etc?

File: tests/integration/cocoon/test_cocoon_instruments.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasdotred) [reply tests/integration/cocoon/test_cocoon_instruments.py:1172]: Including different instruments same portfolio in next push

**Pull request closed without merging**